src/store/flow/block.ts

```diff
@@ -180,7 +180,7 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
 
   block_setLabel({commit, dispatch}, {blockId, value}: {blockId: string, value: string | undefined}) {
     commit('block_setLabel', {blockId, value})
-    dispatch('block_setName', {blockId, value: snakeCaseOnSpaces(value as string)})
+    dispatch('block_setName', {blockId, value: generateBlockCodeFromLabel(value)})
```

Comment:
While the block label can be undefined, it is not necessary to set the name as undefined because even an empty string would trigger a validation error.
(And we are creating blocks with name: '')